### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by petelkavictor**

### CODE REVIEW

File: components/GoogleButton.tsx
Comment (TarasGez) [inline components/GoogleButton.tsx:1]: Can we use SSR? like this:
```
import { signIn } from '@/auth'
import { Button } from '@/components'

function Auth() {
  const login = async () => {
    'use server'
    await signIn('google', { redirectTo: '/profile' })
  }
  return (
    <form action={login}>
      <h1>Login page</h1>
      <Button type="submit">Signin with Google</Button>
    </form>
  )
}
export default Auth
```

File: configs/auth.ts
Comment (TarasGez) [inline configs/auth.ts:1]: We can use it without secret keys:
```
import NextAuth, { NextAuthConfig } from 'next-auth'
import GitHub from 'next-auth/providers/github'
import Google from 'next-auth/providers/google'

export const authOptions: NextAuthConfig = {
  providers: [Google, GitHub],
}

export const { handlers, signIn, signOut, auth } = NextAuth(authOptions)
```
Just add correct names for keys to .env.local:
(https://authjs.dev/getting-started/authentication/oauth)
AUTH_GOOGLE_ID={CLIENT_ID}
AUTH_GOOGLE_SECRET={CLIENT_SECRET}

File: components/ProfileButton.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+'use client'
+import Link from 'next/link'
+import { useSession } from 'next-auth/react'
+
+export default function ProfileButton() {
+  const session = useSession()
```
Comment (TarasGez) [inline components/ProfileButton.tsx:6]: we can use SSR ```const session = await auth()```

**Review by TarasGez — CHANGES_REQUESTED** [review]
Please, add examples of secret keys that you use to the file .env.example in this format
```
AUTH_GOOGLE_ID={CLIENT_ID}
AUTH_GOOGLE_SECRET={CLIENT_SECRET}

AUTH_GITHUB_ID={CLIENT_ID}
AUTH_GITHUB_SECRET={CLIENT_SECRET}
```

**Comment by petelkavictor** [comment]
used NextAuth.js (v4 -  [https://next-auth.js.org](url) ) instead of Auth.js (NextAuth.js v5 - [https://authjs.dev](url) ). Therefore, I close the PR to make authorization on the correct version

**Pull request closed without merging**